### Question 1a)

In [3]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import binom

In [3]:
df = pd.DataFrame(data={'Age of Moose in Years':['Calf (0.5)', '1-5', '6-10', '11-15', '16-20'], 
                        'Number Killed by Wolves': [112, 53, 73, 56, 2]})
df = df.set_index('Age of Moose in Years')
df

Number Killed by Wolves
Age of Moose in Years                         
Calf (0.5)                                 112
1-5                                         53
6-10                                        73
11-15                                       56
16-20                                        2

In [4]:
total = df['Number Killed by Wolves'].sum()
df['Probability of Event'] = df['Number Killed by Wolves'] / total
df

Number Killed by Wolves  Probability of Event
Age of Moose in Years                                               
Calf (0.5)                                 112              0.378378
1-5                                         53              0.179054
6-10                                        73              0.246622
11-15                                       56              0.189189
16-20                                        2              0.006757

### Question 1b)

In [5]:
df['Class Midpoint'] = [0.5, 3, 8, 13, 18]
df

Number Killed by Wolves  Probability of Event  \
Age of Moose in Years                                                  
Calf (0.5)                                 112              0.378378   
1-5                                         53              0.179054   
6-10                                        73              0.246622   
11-15                                       56              0.189189   
16-20                                        2              0.006757   

                       Class Midpoint  
Age of Moose in Years                  
Calf (0.5)                        0.5  
1-5                               3.0  
6-10                              8.0  
11-15                            13.0  
16-20                            18.0

In [6]:
expected_age = (df['Class Midpoint'] * df['Probability of Event']).sum()
print('Expected Age: ', round(expected_age, 2))

Expected Age:  5.28


In [7]:
std = np.sqrt(((df['Class Midpoint'] - expected_age) ** 2 * df['Probability of Event']).sum())
print('Standard Deviation: ', round(std, 4))

Standard Deviation:  4.8754


### Question 2a)

In [13]:
n = 20
r = [18, 19, 20]
p = 0.95

p_5 = binom.pmf(r, n, p).sum()
print('Likelihood of acception: ', round(p_5, 4))

Likelihood of acception:  0.9245


### Question 2b)

In [20]:
p = 0.85

p_15 = binom.pmf(r, n, p).sum()
q_15 = 1 - p_15
print('Likelihood of rejection: ', round(q_15, 4))

Likelihood of rejection:  0.5951


### Question 3a)

In [13]:
n = 1
p = 0.83

p_1 = p * ((1 - p) ** (n-1))
print('Likelihood of passing on the first try: ', p_1)

Likelihood of passing on the first try:  0.83


### Question 3b)

In [14]:
n = [2, 3]

p_2or3 = sum([p * ((1 - p) ** (trial-1)) for trial in n])
print('Likelihood of passing on the second or third try: ', round(p_2or3, 4))

Likelihood of passing on the second or third try:  0.1651


### Question 4a)

In [15]:
mu = 10.75
sigma = 2.2
x = 11

z = (x - mu) / sigma
z

0.11363636363636363

In [16]:
area_right = 1 - stats.norm.cdf(z)
print(f'Likelihood of battery life exceeding {x} hours: ', round(area_right, 4))

Likelihood of battery life exceeding 11 hours:  0.4548


### Question 4b)

In [17]:
x_score = stats.norm.ppf(0.05, loc=mu, scale=sigma)
print('Hours after which you should plan to charge your phone: ', round(x_score, 2))

Hours after which you should plan to charge your phone:  7.13


### Question 5a)

In [18]:
mu = 45
sigma = 12

In [19]:
x1 = 60
z1 = (x1 - mu) / sigma
p_a = 1 - stats.norm.cdf(z1)
round(p_a, 4)

0.1056

In [20]:
x2 = 55
z2 = (x2 - mu) / sigma
p_b = 1 - stats.norm.cdf(z2)
round(p_b, 4)

0.2023

In [21]:
print('Likelihood that cycle time will exceed 60 minutes, given it has already exceeded 55 minutes: ', 
      round(p_a / p_b, 4))

Likelihood that cycle time will exceed 60 minutes, given it has already exceeded 55 minutes:  0.5222


### Question 5b)

In [22]:
x3 = 40
z3 = (x3 - mu) / sigma
p_c = 1 - stats.norm.cdf(z3)
round(p_c, 4)

0.6615

In [23]:
print('Likelihood that cycle time will exceed 55 minutes, given it has already exceeded 40 minutes: ', 
      round(p_b / p_c, 4))

Likelihood that cycle time will exceed 55 minutes, given it has already exceeded 40 minutes:  0.3058


### Question 6a)

In [24]:
mu = 38
sigma = 5
x = 35

In [25]:
z = (x - mu) / sigma
p = stats.norm.cdf(z)
print('Likelihood that drug goes into effect in 35 minutes or less: ', round(p, 4))

Likelihood that drug goes into effect in 35 minutes or less:  0.2743


### Question 6b)

In [26]:
n = 10
sigma_x = sigma / (np.sqrt(n))
sigma_x

1.5811388300841895

In [27]:
z = (x - mu) / sigma_x
p = stats.norm.cdf(z)
print('Likelihood of avg time before the drug is effective for all 10 random patients is 35 minutes or less: ', 
      round(p, 4))

Likelihood of avg time before the drug is effective for all 10 random patients is 35 minutes or less:  0.0289


### Question 6c)

What we see between the two questions above is a difference of 0.2454, or 24.54%, which is a massive difference, especially when considering the second probability was almost 0. This is due to the difference in the standard deviation calculations for the respective problems. The first problem tells us that the standard deviation is 5 minutes, but that changes when we calculate for the random sample size of 10 patients. Here, the standard deviation comes out to about 1.58, which is significantly less. Now the target mark of the drug being effective within 35 minutes has moved from within one standard deviation of the 38-minute mean, to within two standard deviations of the mean, significantly reducing the likelihood that all 10 randomly selected patients will be affected by the drug within 35 minutes. It's intuitive and expected that the likelihood of the drug working within the time frame is greater on any individual trial than it would be for 10 consecutive trials.

### Question 7a)

In [28]:
n = 40
p = 0.85
q = 1 - p

n*p, n*q 

(34.0, 6.000000000000001)

Given the above values for this problem, np and nq are both > 5, therefore it meets the criteria to use normal approximation to this binomial distribution.

### Question 7b)

In [29]:
mu = n * p
sigma = np.sqrt(n * p * q)
print('mu:    ', mu)
print('sigma: ', sigma)

mu:     34.0
sigma:  2.2583179581272432


### Question 7c)

x = r + 0.5
r < 30

P(r < 30) is approximated by P(x < 30.5)

### Question 7d)

In [30]:
x = 30.5
z = (x - mu) / sigma
p = stats.norm.cdf(z)
print('Estimate P(r < 30): ', round(p, 4))

Estimate P(r < 30):  0.0606


According to the calculation above, there is only a probability of 6.06% that this binomial experiment with a success rate of 85% will have fewer than 30 successes out of 40 trials. We can conclude that an outcome of less than 30 successes is highly unlikely.

### Question 8a)

In [31]:
n = 33
p = 0.21
q = 1 - p

n*p, n*q

(6.93, 26.07)

As seen in the above calculations, the np and nq values are both > 5, so we can approximate the  𝑝̂ distribution by a normal distribution.

In [32]:
mu_p = p
sigma_p = np.sqrt(p * n / q)

print('mu_p:    ', mu_p)
print('sigma_p: ', sigma_p)

mu_p:     0.21
sigma_p:  2.9617818789934844


### Question 8b)

In [33]:
n = 25
p = 0.15
q = 1 - p

n*p, n*q

(3.75, 21.25)

As seen directly above, the nq value satisfies the requirement > 5, but the np value falls short. Therefore, we cannot safely approximate the 𝑝̂ distribution by a normal distribution.

### Question 8c)

In [34]:
n = 48

n*p, n*q

(7.199999999999999, 40.8)

As seen in the above calculations, the np and nq values are both > 5, so we can approximate the 𝑝̂ distribution by a normal distribution.

In [36]:
mu_p = p
sigma_p = np.sqrt(p * q / n)

print('mu_p:    ', mu_p)
print('sigma_p: ', sigma_p)

mu_p:     0.15
sigma_p:  0.05153882032022076
